src/main/java/org/opensrp/web/rest/ClientFormResource.java

```diff
@@ -0,0 +1,157 @@
+        ClientFormService.CompleteClientForm completeClientForm = null;
+
+        long formId;
+
+        if (clientFormMetadata == null) {
+            // Get an older form version
+            List<IdVersionTuple> availableFormVersions = clientFormService.getAvailableClientFormMetadataVersionByIdentifier(formIdentifier);
+            DefaultArtifactVersion highestVersion = null;
+            IdVersionTuple highestIdVersionTuple = null;
+
+            for (IdVersionTuple availableFormVersion: availableFormVersions) {
+                DefaultArtifactVersion semanticFormVersion = new DefaultArtifactVersion(availableFormVersion.getVersion());
+
+                if (highestVersion == null || semanticFormVersion.compareTo(highestVersion) > 0) {
+                    highestVersion = semanticFormVersion;
+                    highestIdVersionTuple = availableFormVersion;
+                }
+            }
+
+            if (highestVersion == null) {
+                return new ResponseEntity<>((String) null, HttpStatus.INTERNAL_SERVER_ERROR);
+            } else {
+                formId = highestIdVersionTuple.getId();
+                clientFormMetadata = clientFormService.getClientFormMetadataById(formId);
+            }
+        } else {
+            formId = clientFormMetadata.getId();
+        }
+
+        ClientForm clientForm = clientFormService.getClientFormById(formId);
+        if (clientForm == null) {
+            return new ResponseEntity<>((String) null, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        completeClientForm = new ClientFormService.CompleteClientForm(clientForm, clientFormMetadata);
+
+        return new ResponseEntity<>(gson.toJson(completeClientForm), HttpStatus.OK);
+    }
+    
+    //@RequestMapping(consumes = {MediaType.MULTIPART_FORM_DATA_VALUE}, method = RequestMethod.POST)
+    @RequestMapping(headers = { "Accept=multipart/form-data" }, method = RequestMethod.POST)
+    @ResponseBody
+    private ResponseEntity<String> addClientForm(@RequestParam("form_version") String formVersion,
+                                                 @RequestParam("form_identifier") String formIdentifier,
+                                                 @RequestParam("form_name") String formName,
+                                                 @RequestParam("form") MultipartFile jsonFile,
+                                                 @RequestParam(required = false) String module) {
+        if (TextUtils.isEmpty(formVersion) || TextUtils.isEmpty(formIdentifier) || TextUtils.isEmpty(formName) || jsonFile.isEmpty()) {
+            return new ResponseEntity<>("Required params is empty", HttpStatus.BAD_REQUEST);
+        }
+
+        // TOOD: Check the string is a valid version
+        DefaultArtifactVersion formVersionV = new DefaultArtifactVersion(formVersion);
+        if (!jsonFile.getOriginalFilename().contains(".json")) {
```

Comment:
You should use mime type and not file name to check. Names cannot be trusted to check type of a file.